pdns/recursordist/rec-main.cc

```diff
@@ -2893,6 +2894,35 @@ int main(int argc, char** argv)
     g_slogout = g_slog->withName("out");
 
     ::arg().setSLog(startupLog);
+
+    if (::arg().mustDo("config")) {
+      string config = ::arg()["config"];
+      if (config == "default") {
+        cout << ::arg().configstring(false, true);
+      }
+      else if (config == "check") {
+        try {
+          if (!::arg().file(configname.c_str())) {
+            SLOG(g_log << Logger::Warning << "Unable to open configuration file '" << configname << "'" << endl,
+                 startupLog->error("No such file", "Unable to open configuration file", "config_file", Logging::Loggable(configname)));
+            exit(1);
+          }
+
+          ::arg().parse(argc, argv);
+          exit(0);
+        }
+        catch (const ArgException& argException) {
+          SLOG(g_log << Logger::Warning << "Unable to parse configuration file '" << configname << "'" << endl,
+               startupLog->error("Cannot parse configuration", "Unable to parse configuration file", "config_file", Logging::Loggable(configname)));
```

Comment:
With a wrong config file I now see:
```
Sep 07 09:33:17 msg="Unable to parse configuration file" error="Cannot parse configuration" subsystem="config" level="0" tid="0" ts="1662535997.193" config_file="./recursor.conf"
```
The ectual underlying `argException.reason` is not included in the error message.